Flow.Launcher.Core/Updater.cs

```diff
@@ -29,89 +29,69 @@ public Updater(string gitHubRepository)
             GitHubRepository = gitHubRepository;
         }
 
-        public async Task UpdateApp(IPublicAPI api , bool silentUpdate = true)
+        public async Task UpdateApp(IPublicAPI api, bool silentUpdate = true)
         {
-            UpdateManager updateManager;
-            UpdateInfo newUpdateInfo;
-
-            if (!silentUpdate)
-                api.ShowMsg("Please wait...", "Checking for new update");
-
             try
             {
-                updateManager = await GitHubUpdateManager(GitHubRepository);
-            }
-            catch (Exception e) when (e is HttpRequestException || e is WebException || e is SocketException)
-            {
-                Log.Exception($"|Updater.UpdateApp|Please check your connection and proxy settings to api.github.com.", e);
-                return;
-            }
+                UpdateInfo newUpdateInfo;
+
+                if (!silentUpdate)
+                    api.ShowMsg("Please wait...", "Checking for new update");
+
+                using var updateManager = await GitHubUpdateManager(GitHubRepository);
+
 
-            try
-            {
                 // UpdateApp CheckForUpdate will return value only if the app is squirrel installed
                 newUpdateInfo = await updateManager.CheckForUpdate().NonNull();
-            }
-            catch (Exception e) when (e is HttpRequestException || e is WebException || e is SocketException)
-            {
-                Log.Exception($"|Updater.UpdateApp|Check your connection and proxy settings to api.github.com.", e);
-                updateManager.Dispose();
-                return;
-            }
 
-            var newReleaseVersion = Version.Parse(newUpdateInfo.FutureReleaseEntry.Version.ToString());
-            var currentVersion = Version.Parse(Constant.Version);
+                var newReleaseVersion = Version.Parse(newUpdateInfo.FutureReleaseEntry.Version.ToString());
+                var currentVersion = Version.Parse(Constant.Version);
 
-            Log.Info($"|Updater.UpdateApp|Future Release <{newUpdateInfo.FutureReleaseEntry.Formatted()}>");
+                Log.Info($"|Updater.UpdateApp|Future Release <{newUpdateInfo.FutureReleaseEntry.Formatted()}>");
 
-            if (newReleaseVersion <= currentVersion)
-            {
-                if (!silentUpdate)
-                    MessageBox.Show("You already have the latest Flow Launcher version");
-                updateManager.Dispose();
-                return;
-            }
+                if (newReleaseVersion <= currentVersion)
+                {
+                    if (!silentUpdate)
+                        MessageBox.Show("You already have the latest Flow Launcher version");
+                    updateManager.Dispose();
```

Comment:
since it's using `using var updateManager = await GitHubUpdateManager(GitHubRepository);` in the above code, this Dispose() is not needed right?